examples/wandb_sweeps/sweep.py

```diff
@@ -41,10 +41,12 @@ def run_sweep(argv: List[str]) -> None:
     We encapsulate each run by using a separate subprocess, which ought to
     ensure that memory is returned (etc.).
     """
-    try:
-        subprocess.check_call(argv)
-    except subprocess.CalledProcessError as error:
-        logging.error("Subprocess error: %s", error)
+    process = subprocess.Popen(argv, stderr=subprocess.PIPE, text=True)
+    for line in process.stderr:
+        logging.info(line.rstrip())
+    return_code = process.wait()
```

Comment:
My last nit seems to have been swallowed by GitHub so I'm trying again. I see why you were confused!

So right now the `info` log is just the stderr from the subprocess, and that seems fine to me. I think adding the explicit exit code to the log is not necessary, since I confirmed (by looking at the W&B source) that it already logs the exit code (which is why `finish` takes that as an argument) at a higher level. So I'd remove line 48 and combine lines 47 and 49 into one, and that's my last ask. 

In general---and reasonable people may disagree with this---I prefer:

```python
finalize(foo(...))
```

to

```python
bar = foo(...)
finalize(bar)
```

except when:

1. the right-hand side (RHS) expression I'm abbreviating `foo(...)` is unusually long or complex, or 
2. (relatedly) it is hard to reason from context what `foo(...)` is, exactly, and the variable name gives the reader some context, or
3. the RHS will be used again in the same scope so it's sensible to save it somewhere.
4. (relatedly) there is some reason to override the variable.

Here I'd say it may not be immediately obvious to the reader that `wait` returns the exit code (even if that is the documented API for this built-in module), so (2) is possibly applicable, but

```python
wandb.finish(exit_code=process.wait())
```

makes it reasonably obvious what's going on and negates the benefit.